sqlglot/parser.py

```diff
@@ -5475,6 +5484,44 @@ def _parse_comprehension(
             condition=condition,
         )
 
+    def _parse_heredoc(self) -> t.Optional[exp.Heredoc]:
+        tags = []
+        tag_text = None
+
+        if self._match_text_seq("$"):
+            tags.append("$")
+        else:
+            return None
+
+        if self._curr and self._is_connected():
+            if not self._match_text_seq("$"):
+                tag_text = self._curr.text
+                tags.append(tag_text.upper())
+                self._advance()
+            else:
+                tags.append("$")
+        else:
+            self.raise_error("No closing $ found")
+
+        if tag_text:
```

Comment:
And so instead of checking `tag_text` here, the proxy would be if the last tag appended is _not_ a single dollar?